cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -1766,14 +1812,15 @@ struct IsTitleUnicode {
                                [&previous_cased, &rule_3](uint32_t codepoint) {
                                  if (IsLowerCaseCharacterUnicode(codepoint)) {
                                    if (!previous_cased) return false;  // rule 1 broken
+                                   // next should be more lower case or uncased
                                    previous_cased = true;
-                                 } else if (IsCasedCharacterUnicode(codepoint)) {
+                                 } else if (IsUpperCaseCharacterUnicode(codepoint)) {
```

Comment:
Not sure why you changed this. Does it make a difference ("cased and not lowercase" should be the same as "uppercase", probably) or is it only for symmetry?

Reply:
Just for symmetry but I will revert this change as the cased checking is a bit simpler.